contracts/BadAuth.sol

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Peter Robinson 2023
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
```

Comment:
This is good stuff. We should not use too old solidity version for testing. Current version is 0.8.23, I guess anything > 0.8 is good.